gammapy/makers/utils.py

```diff
@@ -248,18 +326,31 @@ def make_map_background_irf(
     else:
         image_geom = geom.to_image()
         d_omega = image_geom.solid_angle()
-
-    coords = _get_fov_coords(
-        pointing=pointing,
-        irf=bkg,
-        geom=geom,
-        use_region_center=use_region_center,
-        obstime=obstime,
-    )
-    coords["energy"] = broadcast_axis_values_to_geom(geom, "energy", False)
-
-    bkg_de = bkg.integrate_log_log(**coords, axis_name="energy")
-    data = (bkg_de * d_omega * ontime).to_value("")
+    if bkg.has_offset_axis or bkg.fov_alignment == FoVAlignment.RADEC:
+        data = _evaluate_bkg(
+            pointing, ontime, bkg, geom, use_region_center, obstime, d_omega
+        )
+    if bkg.fov_alignment == FoVAlignment.ALTAZ:
+        endtime = obstime + ontime
+        data = np.zeros(geom.data_shape)
+        while obstime < endtime:
+            # Evaluate the step time needed to have a FoV rotation of fov_rotation_step
+            # Minimum step of 1 second to avoid infinite computation very close to zenith
+            steptime = max(
+                compute_rotation_time_step(
+                    fov_rotation_step, pointing.get_altaz(obstime)
+                ),
+                MINIMUM_TIME_STEP,
+            )
+            steptime = (
+                steptime
+                if steptime < endtime - obstime - MINIMUM_TIME_STEP
+                else endtime - obstime
+            )
+            data += _evaluate_bkg(
+                pointing, steptime, bkg, geom, use_region_center, obstime, d_omega
+            )
+            obstime = obstime + steptime
```

Comment:
It looks to me a bit strange to modify a function argument. I would have used a dedicated variable

Reply:
Ok I will change this.